beetsplug/discogs.py

```diff
@@ -180,6 +180,55 @@ def candidates(self, items, artist, album, va_likely, extra_tags=None):
             self._log.debug('Connection error in album search', exc_info=True)
             return []
 
+    def item_candidates(self, item, artist, title):
+        """Returns a list of TrackInfo objects for Search API results
+        matching ``title`` and ``artist``.
+        :param item: Singleton item to be matched.
+        :type item: beets.library.Item
+        :param artist: The artist of the track to be matched.
+        :type artist: str
+        :param title: The title of the track to be matched.
+        :type title: str
+        :return: Candidate TrackInfo objects.
+        :rtype: list[beets.autotag.hooks.TrackInfo]
+        """
+        if not self.discogs_client:
+            return
+
```

Comment:
Around here you could do an early exit, similar to that one in `candidates`: https://github.com/ghbrown/beets/blob/discogs_s_ic/beetsplug/discogs.py#L161-L163
It would address your legimite concerns described in "Known fragility"

Back then, when I submitted this early exit in `candidates` I was not sure if such an exit should also happen when _only_ artist is given. As you mentionend, results are pretty bad and almost useless then. But the good thing is, in a verbose output a user would see something like this, which tells that album is missing and only artist is considered in the search: `Search terms: Dave Clarke - Remixes & Rarities 1992-2005` (happening in `beets.autotag.match.py`)

Since something similar is coming from that module: `Item search terms: <artist> <title>`, I think an early exit if both artist and title are missing it is sufficient and most of all streamlines the behaviour with regular album discogs search.

Reply:
Thanks for the suggestion. I agree, I think early exiting only when both track name and artist are missing is a better choice for now.